scandium-core/src/main/java/org/eclipse/californium/scandium/util/SecretUtil.java

```diff
@@ -0,0 +1,232 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Bosch Software Innovations GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ *
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial implementation
+ ******************************************************************************/
+package org.eclipse.californium.scandium.util;
+
+import java.security.MessageDigest;
+import java.security.spec.KeySpec;
+import java.util.Arrays;
+
+import javax.crypto.SecretKey;
+import javax.security.auth.DestroyFailedException;
+import javax.security.auth.Destroyable;
+
+import org.eclipse.californium.elements.util.Bytes;
+
+/**
+ * Extracts psk credentials from the current {@code DTLSSession}.
+ */
+public class SecretUtil {
+
+	/**
+	 * Destroy secret key.
+	 * 
+	 * {@link SecretKey#equals(Object)} and {@link SecretKey#hashCode} must not
+	 * be used after the key is destroyed!
+	 * 
+	 * @param key secret key to destroy. If {@code null}, nothing is destroyed.
+	 */
+	public static void destroy(SecretKey key) {
+		if (key instanceof Destroyable) {
+			destroy((Destroyable) key);
+		}
+	}
+
+	/**
+	 * Destroy provided security destroyable.
+	 * 
+	 * @param clean object to destroy. Maybe {@code null}.
+	 */
+	public static void destroy(Destroyable clean) {
+		if (clean != null) {
+			try {
+				clean.destroy();
+			} catch (DestroyFailedException e) {
```

Comment:
DEBUG?
I think, it should be tested during development, if it fits the expectations and requirements.

Reply:
With events which should never happened, I think raise a WARN which explain that should never happen is pretty much a good practice.

Reply:
FMPOV there are two case:

- This should never happen, but it would otherwise harm seriously
- This should never happen, and it's simply not worth to spend too much time in it.

For me it's simply the second. But though switch to warn would save also time, why not.